src/components/buttons/newsletter-button.js

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import useNewsLetterSubscriptionAPI from "../../hooks/use-newsletter-subscription"
+import toast, { Toaster } from 'react-hot-toast';
+
+export default function NewsletterSubscriptionButton(props) {
+  const { subscribe, check_subscription_status } = useNewsLetterSubscriptionAPI();
+  const notify = (value) => toast(value);
+  const error = (value) => toast.error(value);
+  const loading = () => toast.loading("loading");
+  const success = (value) => toast.success(value);
+  return (
+    <button
+      className="button dark-text locked light-text"
+      onClick={(e) => {
+        
+        check_subscription_status(props.user_id).then((subscribed) => {
+          if(subscribed == "Login required"){
+            error("Login Required")
+          }
```

Comment:
Sometimes the return value from `check_subscription_status` status is the error string, esp when the user hasn't logged in before subbing or unsubbing.